### DISCUSSION TITLE
Capstone Project Subtask - Banking Systems

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/spring-framework#23] [Tasks]
1. Create the Admin and Employee models

2. Create the DB according to the given diagram.

3. Come up with functions to be included in the `EmployeeController` and the `AdminController`.

4. Create route URLs and payloads to test all the controllers.

**Comment by Mrafe** [comment]
###  Employee Controller:

- 	showCustomers();
- 	updateCustomer();
- 	createCustomer();
- 	deleteCustomer();
- 	showEmployee();
- 	updateEmployeeProfile();
- 	showTransactions();

### **Admin Controller:**

- showCustomers();
- 	updateCustomer();
- 	createCustomer();
- 	deleteCustomer();
- 	showEmployee();
- 	updateEmployeeProfile();
- 	creatEmployee();
- 	deleteEmployee();
- 	updateAdminProfile();
- 	showAdminProfile();
- 	showTransactions();

**Comment by Gaurav-Joshi-31** [comment]
Admin--

- employee CRUD operations
- showAllTransactionsDetails
- view Admin profile
- 
Employee-
- showCustomerDetails
- view employee profile

**Comment by aish955** [comment]
**Employee Controller:**


    CRUD for customers:
    showCustomers();
    updateCustomer();
    createCustomer();
    deleteCustomer();

    showEmployee(); 
    updateEmployeeProfile();
    showTransactions();

**Admin Controller:**

    CRUD for customers and employees:
    showCustomers();
    updateCustomer();
    createCustomer();
    deleteCustomer();

    showEmployee();
    createEmployee();
    deleteEmployee();
    updateEmployeeProfile();

    showAdminProfile();
    updateAdminProfile();
    showTransactions();

**Comment by Shubhi27** [comment]
### Admin:-
```
- createEmployee
- updateEmployeeDetails
- showEmplyoeesDetails
- deleteEmployee
- showAllTransactionsDetails
- showCustomerDetails
- showAdminProfileDetails
```


### Employee:-
```
- showCustomerDetails
- showTransactionDetails
- showEmployeeDetails
```

**Comment by shreiyarandive** [comment]
**Employee Controller:**
```
showCustomers();
updateCustomer();
createCustomer();
deleteCustomer();
showEmployee();
updateEmployeeProfile();
showTransactions();
```

**Admin Controller:**

```showCustomers();
updateCustomer();
createCustomer();
deleteCustomer();
showEmployee();
updateEmployeeProfile();
showAdminProfile();
updateAdminProfile();
createEmployee();
deleteEmployee();
showTransactions();

**Comment by Batchusatyasri** [comment]
### SUBTASK 2

**Employee Controller**
```
createCustomers();
deleteCustomers();
updateCustomers();
issueLoanToCustomers();
updateEmployeeProfile();
showTransactions();
```

**AdminController**
 
```
createEmployee();
updateEmployee();
deleteEmployee();
creditSalaryToEmployee();
showTransactions();
```

**Comment by mkaifahmed** [comment]
**Employee Controller:**

    ```
    showCustomers();
    updateCustomer();
    createCustomer();
    deleteCustomer();
    showEmployee();
    updateEmployeeProfile();
    showTransactions();
```

**Admin Controller:**

```
    showCustomers();
    updateCustomer();
    createCustomer();
    deleteCustomer();
    showTransactions();
    showEmployee();
    updateEmployeeProfile();
    createEmployee();
    deleteEmployee();
```

**Comment by Amrit-Raj22** [comment]
**Employee controller**

```
createCustomer();
updateCustomer();
deleteCustomer();
showCustomers();
showEmployee();
updateEmployeeProfile();
showTransactions();
```

**Admin controller**

```
showCustomers();
createCustomer();
updateCustomer();
deleteCustomer();
showEmployee();
createEmployee();
updateEmployeeProfile();
deleteEmployee();
showAdminProfile();
updateAdminProfile();
showTransactions();
```

**Comment by ratnadeepikavuddagiri** [comment]
**Employee Controller**
insertCustomer()
updateCustomer()
deleteCustomer()
viewCustomer()
viewAllCustomers()
viewEmployeeProfile()
updateEmployee()
viewTransactions()

**Admin Controller**
insertEmployee()
updateEmployee()
deleteEmployee()
viewEmployees()
viewTransactions()
viewCustomers()
viewAdminProfile()
updateAdmin()
insertAdmin()

**Comment by preeti-max** [comment]
EmployeeController

        createCustomer() POST
	showCustomers() GET
	updateCustomer() PATCH
	deleteCustomer() DELETE
	showEmployeeDetails() GET
	updateEmployeeDetails() PATCH
	showTransactions() GET
AdminController

        createCustomer() POST
	showCustomers() GET
	updateCustomer() PATCH
	deleteCustomer() DELETE
	showAdminProfileDetails() GET
        updateAdminProfile() PATCH
	showEmployees() GET
	createEmployee() POST
	updateEmployeeDetails() PATCH
	deleteEmployee() DELETE
	showTransactions() GET

**Comment by DivikaAgrawal** [comment]
**Employee Controller:**


createCustomer();
showCustomers();
createEmployee();
showEmployeeProfile();
updateEmployeeProfile();
creatEmployee();
deleteEmployee();
showBranchTransactions();


**Admin Controller:**

createCustomer();
showAllCustomers();
deleteCustomer();
showAdminProfile();
showTransactions();

**Comment by Sarfinaa** [comment]
**Admin Controller**
- addEmployee();
- updateEmployeeDetails();
- showEmployeeDetails();
- showAllEmployeeDetails();
- deleteEmployee();
- showAllTransactionsDetails();
- showCustomerDetails();

**Employee Controller**

- showCustomers();
- updateCustomer();
- createCustomer();
- deleteCustomer();
- getCustomer();
- showCustomerTransactions();

**Reply by Sarfinaa** [reply]
@akash-coded sir, I have some doubts about the project.

**Reply by akash-coded** [reply]
@Sarfinaa Yeah, tell me about it.

**Reply by Sarfinaa** [reply]
@akash-coded  sir, why do we need a bank table as a separate from the branch table like we can add bank name column in branch table then why do we need it?

**Reply by akash-coded** [reply]
@Sarfinaa Yes, you can add the bank name as a `varchar(60)` column to the `branches` table, and consequently as a `String` data member in the `Branch` model class. Banks will be kept in a separate table in case a bank contains more fields than just its name, such as the manager's name, address, etc. In that case, the `branches` table will have a foreign key to the primary key of the `banks` table. And the `Branch` model class will have a dependency (object reference) of the `Bank` class.

**Comment by SrideviYamala** [comment]
### Employee Controller:

CRUD operations of Customer-
createCustomer();
showCustomers();
updateCustomer();
deleteCustomer();

showEmployeeProfile();
updateEmployeeProfile();
showTransactions();

### Admin Controller:

CRUD operations of Customer-
createCustomer();
showCustomers();
updateCustomer();
deleteCustomer();

CRUD operations of Employee
creatEmployee();
showEmployees();
updateEmployeeProfile();
deleteEmployee();

showAdminProfile();
updateAdminProfile();
showTransactions();

**Comment by hemanth409** [comment]
Employee Controller:

showCustomers();
updateCustomer();
createCustomer();
deleteCustomer();
showEmployee();
updateEmployeeProfile();
showTransactions();

Admin Controller:

showCustomers();
updateCustomer();
createCustomer();
deleteCustomer();
showEmployee();
updateEmployeeProfile();
showAdminProfile();
updateAdminProfile();
createEmployee();
deleteEmployee();
showTransactions();

**Comment by RohitN09** [comment]
### Employee Controller:

```
showCustomers();
updateCustomer();
createCustomer();
deleteCustomer();
showEmployee();
updateEmployeeProfile();
showTransactions();
```

### Admin Controller:

```
showAllCustomers();
updateCustomer();
createCustomer();
deleteCustomer();
showEmployee();
updateEmployeeProfile();
showAdminProfile();
createEmployee();
deleteEmployee();
showTransactions();
```

**Comment by Ashwesh09** [comment]
AdminController:

- createEmployee();
- updateEmployeeDetails();
- deleteEmployee();
- showEmployeeDetails();
- showAllEmployeeDetails();
- showAllTransactionsDetails();
- showCustomerDetails();
- showAllCustomerDetails();


EmployeeController:

- createCustomer();
- updateCustomer();
- deleteCustomer();
- showCustomer();
- showAllCustomers();
- showAllTransactionsDetails();

**Comment by Harshita-Joshi** [comment]
### EmployeeController

- createCustomer()
- getCustomer()
- updateCustomer()
- deleteCustomer()
- getCustomers()
- showTransactionsList()


### AdminController

-  createCustomer()
- getCustomer()
- updateCustomer()
- deleteCustomer()
-  createEmployee()
- getEmployee()
- updateEmployee()
- deleteEmployee()
- getEmployees()
- showTransactionsList()

**Comment by lalit-ky** [comment]
### Employee Controller:

```
CRUD for customers:
showCustomers();
updateCustomer();
createCustomer();
deleteCustomer();

showEmployee(); 
updateEmployeeProfile();
showTransactions();
```

### Admin Controller:
```

CRUD for customers and employees:
showCustomers();
updateCustomer();
createCustomer();
deleteCustomer();

showEmployee();
createEmployee();
deleteEmployee();
updateEmployeeProfile();

showAdminProfile();
updateAdminProfile();
showTransactions();
```

**Comment by KaminiJha** [comment]
## Employee
showCustomers();
updateCustomer();
createCustomer();
deleteCustomer();
showEmployee();
showTransactions();

## Admin
showCustomers();
updateCustomer();
createCustomer();
deleteCustomer();
showEmployee();
creatEmployee();
deleteEmployee();
updateAdminProfile();
showAdminProfile();
showTransactions();

**Comment by rakeshGit419** [comment]
**Employee Controller:**
showCustomers();
updateCustomer();
createCustomer();
deleteCustomer();
showEmployee();
updateEmployeeProfile();
showTransactions();


**Admin Controller:**
showCustomers();
updateCustomer();
createCustomer();
deleteCustomer();
showEmployee();
updateEmployeeProfile();
creatEmployee();
deleteEmployee();
updateAdminProfile();
showAdminProfile();
showTransactions();

**Comment by Prem-3** [comment]
**Employee Controller:**
showCustomers();
updateCustomer();
createCustomer();
deleteCustomer();
showEmployee();
updateEmployeeProfile();
showTransactions();

**Admin Controller:**
showCustomers();
updateCustomer();
createCustomer();
deleteCustomer();
showEmployee();
creatEmployee();
deleteEmployee();
updateAdminProfile();
showAdminProfile();
showTransactions();
